crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -409,6 +399,22 @@ impl RoomEventCacheInner {
             all_duplicates,
         ) = state.collect_valid_and_duplicated_events(timeline.events).await?;
 
+        // If the timeline isn't limited, and we already knew about some past events,
+        // then this definitely know what the timeline head is (either we know
+        // about all the events persisted in storage, or we have a gap
+        // somewhere). In this case, we can ditch the previous-batch
+        // token, which is an optimization to avoid unnecessary future back-pagination
+        // requests.
+        //
+        // We can also ditch it, if we knew about all the events that came from sync,
```

Comment:
Don't need a comma after `it` as the first part cannot stand on its own as a sentence.